src/tests/unit/common/rule-based-view-model-provider.test.ts

```diff
@@ -1,104 +1,57 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+import { CardSelectionViewData } from 'common/get-card-selection-view-data';
+
 import { getUnifiedRuleResults } from '../../../common/rule-based-view-model-provider';
-import { CardRuleResult, CardRuleResultsByStatus } from '../../../common/types/store-data/card-view-model';
+import { CardResult, CardRuleResultsByStatus } from '../../../common/types/store-data/card-view-model';
 import { InstanceResultStatus, UnifiedResult, UnifiedRule } from '../../../common/types/store-data/unified-data-interface';
 
 describe('RuleBasedViewModelProvider', () => {
+    const emptyCardSelectionViewData = {} as CardSelectionViewData;
+
     test('getUnifiedRuleResults with null rules and results', () => {
-        const actualResults: CardRuleResultsByStatus = getUnifiedRuleResults(null, null);
+        const actualResults: CardRuleResultsByStatus = getUnifiedRuleResults(null, null, null);
 
         expect(actualResults).toEqual(null);
     });
 
     test('getUnifiedRuleResults with null rules', () => {
-        const actualResults: CardRuleResultsByStatus = getUnifiedRuleResults(null, []);
+        const actualResults: CardRuleResultsByStatus = getUnifiedRuleResults(null, [], emptyCardSelectionViewData);
 
         expect(actualResults).toEqual(null);
     });
 
     test('getUnifiedRuleResults with null results', () => {
-        const actualResults: CardRuleResultsByStatus = getUnifiedRuleResults([], null);
+        const actualResults: CardRuleResultsByStatus = getUnifiedRuleResults([], null, emptyCardSelectionViewData);
+
+        expect(actualResults).toEqual(null);
+    });
+
+    test('getUnifiedRuleResults with null card selection view data', () => {
+        const actualResults: CardRuleResultsByStatus = getUnifiedRuleResults([], [], null);
 
         expect(actualResults).toEqual(null);
     });
 
     test('getUnifiedRuleResults', () => {
```

Comment:
I think the `isSelected`/`isExpanded` detection in here is the most interesting bit of logic being added by this PR; should there be tests that cover those different combinations?

Reply:
Updated tests.